src/ToolBox/SOS/Strike/strike.cpp

```diff
@@ -14019,4 +14019,4 @@ Help(PDEBUG_CLIENT Client, PCSTR Args)
     return S_OK;
 }
 
-#endif // !FEATURE_PAL
+#endif // FEATURE_PAL
```

Comment:
Yes, it is a big mess.  I’ll try to clean this up in the future.

From: Aditya Mandaleeka [mailto:notifications@github.com]
Sent: Monday, August 3, 2015 12:15 PM
To: dotnet/coreclr <coreclr@noreply.github.com>
Cc: Mike McLaughlin <mikem@microsoft.com>
Subject: Re: [coreclr] Add work around a bug in lldb on a core dump (#1327)


In src/ToolBox/SOS/Strike/strike.cpp<https://github.com/dotnet/coreclr/pull/1327#discussion_r36120010>:

> @@ -14019,4 +14019,4 @@ Help(PDEBUG_CLIENT Client, PCSTR Args)

>      return S_OK;

>  }

>

> -#endif // !FEATURE_PAL

> +#endif // FEATURE_PAL

Not necessarily something to address in this PR, but it looks like there's some hairy #ifdef nesting going on in this file. It took me a while to find the matching #ifndef FEATURE_PAL (I think it's the one at line 12660).

The other issue with it is that there is some inconsistency with what the comments next to the endif's say when they're closing an #ifndef. If there's #ifndef FOO, sometimes the endif is #endif // FOO and elsewhere it's #endif // !FOO. This makes those comments almost useless since you end up having to scroll up to figure out what's going on anyway.

Again, this doesn't need to be fixed in this PR, but I think it would be good to standardize that and also detangle some of the nesting in this file.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/1327/files#r36120010>.